server/valgrind-out.txt

```diff
@@ -1,150 +1,157 @@
+--78762-- REDIR: 0x4924650 (libc.so.6:__strrchr_sse2) redirected to 0x484e870 (__strrchr_sse2)
+--78762-- REDIR: 0x490f0a0 (libc.so.6:malloc) redirected to 0x4848820 (malloc)
+--78762-- REDIR: 0x4910520 (libc.so.6:calloc) redirected to 0x484d9d0 (calloc)
+--78762-- REDIR: 0x4924440 (libc.so.6:__strchrnul_sse2) redirected to 0x4853330 (strchrnul)
+--78762-- REDIR: 0x4924950 (libc.so.6:__strlen_sse2) redirected to 0x484ed70 (__strlen_sse2)
+--78762-- REDIR: 0x490f3e0 (libc.so.6:free) redirected to 0x484b210 (free)
+--78762-- REDIR: 0x492e870 (libc.so.6:memcpy@GLIBC_2.2.5) redirected to 0x484fe00 (memcpy@GLIBC_2.2.5)
+--78762-- REDIR: 0x4912fc0 (libc.so.6:__GI_strstr) redirected to 0x4853ae0 (__strstr_sse2)
+--78762-- REDIR: 0x4a0cd30 (libc.so.6:__memcmp_sse4_1) redirected to 0x4851f80 (__memcmp_sse4_1)
+--78762-- REDIR: 0x491cd70 (libc.so.6:__strcmp_sse2_unaligned) redirected to 0x484fbd0 (strcmp)
+--78762-- REDIR: 0x490f740 (libc.so.6:realloc) redirected to 0x484dc50 (realloc)
+--78762-- REDIR: 0x492ef60 (libc.so.6:__memset_sse2_unaligned) redirected to 0x4852770 (memset)
+--78762-- REDIR: 0x492e850 (libc.so.6:__mempcpy_sse2_unaligned) redirected to 0x4853440 (mempcpy)
+--78762-- REDIR: 0x4929890 (libc.so.6:__strcpy_sse2_unaligned) redirected to 0x484ee60 (strcpy)
+--78762-- REDIR: 0x491e880 (libc.so.6:__memchr_sse2) redirected to 0x484fd50 (memchr)
+==78762== 
+==78762== HEAP SUMMARY:
+==78762==     in use at exit: 1,160 bytes in 7 blocks
+==78762==   total heap usage: 253 allocs, 246 frees, 617,263 bytes allocated
+==78762== 
+==78762== Searching for pointers to 7 not-freed blocks
+==78762== Checked 33,655,336 bytes
+==78762== 
+==78762== 72 bytes in 3 blocks are still reachable in loss record 1 of 3
+==78762==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
+==78762==    by 0x109FEC: main (aesdsocket.c:412)
+==78762== 
+==78762== 272 bytes in 1 blocks are possibly lost in loss record 2 of 3
+==78762==    at 0x484DA83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
+==78762==    by 0x40147D9: calloc (rtld-malloc.h:44)
+==78762==    by 0x40147D9: allocate_dtv (dl-tls.c:375)
+==78762==    by 0x40147D9: _dl_allocate_tls (dl-tls.c:634)
+==78762==    by 0x48FF7B4: allocate_stack (allocatestack.c:430)
+==78762==    by 0x48FF7B4: pthread_create@@GLIBC_2.34 (pthread_create.c:647)
+==78762==    by 0x109F28: main (aesdsocket.c:385)
+==78762== 
+==78762== 816 bytes in 3 blocks are possibly lost in loss record 3 of 3
+==78762==    at 0x484DA83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
+==78762==    by 0x40147D9: calloc (rtld-malloc.h:44)
+==78762==    by 0x40147D9: allocate_dtv (dl-tls.c:375)
+==78762==    by 0x40147D9: _dl_allocate_tls (dl-tls.c:634)
+==78762==    by 0x48FF7B4: allocate_stack (allocatestack.c:430)
+==78762==    by 0x48FF7B4: pthread_create@@GLIBC_2.34 (pthread_create.c:647)
+==78762==    by 0x10A059: main (aesdsocket.c:424)
```

Comment:
These warnings are telling you that you aren't handling cleanup for threads which haven't completed in your main loop before your signal handler exits.